src/services/itemMembership/repository.ts

```diff
@@ -34,14 +35,57 @@ import { mapById } from '../utils';
 import { ItemMembership } from './entities/ItemMembership';
 import { getPermissionsAtItemSql } from './utils';
 
-export const ItemMembershipRepository = AppDataSource.getRepository(ItemMembership).extend({
+type ItemPath = Item['path'];
+type AccountId = Account['id'];
+type CreatorId = Account['id'];
+
+type CreateItemMembershipBody = {
+  itemPath: ItemPath;
+  accountId: AccountId;
+  creatorId?: CreatorId;
+  permission: PermissionLevel;
+};
+type UpdateItemMembershipBody = { permission: PermissionLevel };
+type KeyCompositionItemMembership = { itemPath: ItemPath; accountId: AccountId };
+type ResultMoveHousekeeping = {
+  inserts: CreateItemMembershipBody[];
+  deletes: KeyCompositionItemMembership[];
+};
+
+export class ItemMembershipRepository extends MutableRepository<
+  ItemMembership,
+  UpdateItemMembershipBody
+> {
+  constructor(manager?: EntityManager) {
+    super(DEFAULT_PRIMARY_KEY, ItemMembership, manager);
+  }
+
+  async getOne(id: string): Promise<ItemMembership | null> {
+    return await super.findOne(id, {
+      relations: {
+        creator: true,
+        account: true,
+        item: true,
+      },
+    });
+  }
+
   /**
    * Create multiple memberships given an array of partial membership objects.
    * @param memberships Array of objects with properties: `memberId`, `itemPath`, `permission`, `creator`
    */
-  async createMany(memberships: Partial<ItemMembership>[]): Promise<ItemMembership[]> {
-    return this.insert(memberships);
-  },
+  async addMany(memberships: CreateItemMembershipBody[]): Promise<ItemMembership[]> {
+    const itemsMemberships = memberships.map((m) =>
+      this.repository.create({
+        permission: m.permission,
+        item: { id: getChildFromPath(m.itemPath), path: m.itemPath },
+        account: { id: m.accountId },
+        creator: { id: m.creatorId },
+      }),
+    );
+    await this.repository.insert(itemsMemberships);
+    return itemsMemberships;
```

Comment:
When using the `create` method, is the id generated too?

Reply:
I'm not sure about which `id` you're talking about.
If you're thinking about the ItemMembership ID, then no. It will be generated on the database.